scripts/reporting/statsGenElastic/importOldStats.js

```diff
@@ -0,0 +1,174 @@
+const ElasticClient = new Client({
+	cloud: {
+	  id: process.env.ELASTIC_CLOUD_ID
+	},
+	auth: {
+	  username: ELASTIC_CLOUD_AUTH[0],
+	  password: ELASTIC_CLOUD_AUTH[1]
+	},
+	reload_connections: true,
+	maxRetries: 5,
+	request_timeout: 60
+  });
+  
+ElasticClient.cluster.health({},function(err,resp,status) {  
+	console.log("[ELASTIC] -- Client Health --",resp);
+});
+
+if(process.argv.length < 5) {
+	console.error('Not enough arguments: \n\tUsage: node importOldStats.js <mongo host:port> <username> <password> <import-csv-folder>');
+	process.exit(-1);
+}
+
+try {
+	if ( Utils.clean(process.argv[5] === undefined ) ) { errors.push(new Error('import-csv-folder is not set')); } 
+	if ( errors.length > 0 ) { throw errors }
+} catch (e) {
+    for (let i=0; i<e.length; i++) {
+		console.log(e[i].message);
+	  }
+	process.exit(-1);
+}
+
+const url = `mongodb://${process.argv[3]}:${process.argv[4]}@${process.argv[2]}/admin`;
+const client = new MongoClient(url, { useUnifiedTopology: true });
+const importFolder = process.argv[5]
+let opt  = {delimiter: ',', quote: '\'', escape: '\'', relax: true, skip_empty_lines: true};
+
+start();
+
+async function start() {
+	console.log(`Trying to connect to ${url}...`);
+	try {
+		const db = await client.connect();
+		console.log('Connected successfully!');
+
+		(async ()=>{
+			try {
+				// Get the files as an array
+				const files = await fs.promises.readdir( importFolder );
+				const seen = [] // array to catch previously seen id's
+
+				for( const file of files ) {
+					const fullPath = path.join( importFolder, file );
+					const stat = await fs.promises.stat( fullPath );
+					
+					if( stat.isFile() && file.includes(".csv") && !file.includes("com.dropbox.attrs")  ){
+						console.log( "'%s' is a csv file.", fullPath );
+
+						fs.createReadStream(fullPath)
+						.pipe(csv({
+								headers: ["Teamspace","LastLogin"]
+								}
+							))
+						.on('data', (data) => {
+							if ( !Utils.isUndefined(data.Teamspace) && !Utils.isUndefined( Utils.clean(data.LastLogin) ) ) {
+
+								const lastLogin = {
+									"Teamspace" : String(data.Teamspace),
+									"Last Login" : String(data.LastLogin),
+									"DateTime" : Utils.formatDate(new Date(data.LastLogin))
+								}
+
+								const id = Utils.hashCode( Object.values(lastLogin).toString() )
+
+								if (!seen.includes(id) ) {
+									const request = ElasticClient.search({
+										index: Utils.teamspaceIndexPrefix + "-login",
+										body: {
+											query: {
+												match: { id: id }
+											}
+										}
+									}, {
+										ignore: [404],
+										maxRetries: 3
+									})
+									request
+									.then(result => {
+										if (result.body.hits.total.value === 0) {
+												console.log("creating",seen.length,":::",id)
+												Utils.createElasticRecord( ElasticClient, Utils.teamspaceIndexPrefix + "-login", lastLogin, id)
+												seen.push(id) // log hash is created to save calls
```

Comment:
avoid duplication, move seen.push(id) outside of the if condition since you're pushing it anyway.

Reply:
done